src/ciflow-bot.ts

```diff
@@ -475,11 +475,21 @@ export class Ruleset {
       body = `${commentBody}\n${ciflowCommentStart}${body}${ciflowCommentEnd}`;
     }
 
-    await this.ctx.github.issues.updateComment({
-      body,
-      owner: this.owner,
-      repo: this.repo,
-      comment_id: commentID
-    });
+    if (commentId === 0) {
+      const res = await this.ctx.github.issues.createComment({
+        body,
+        owner: this.owner,
+        repo: this.repo,
+        issue_number: this.pr_number
+      });
+      commentId = res.data.id;
+    } else {
+      await this.ctx.github.issues.updateComment({
+        body,
+        owner: this.owner,
+        repo: this.repo,
+        comment_id: commentId
+      });
```

Comment:
This would update the comment even if there are no changes in the body, does it?

Reply:
Yes, the comment will change when the 

1) new comment command instructs the bot, or
2) PR synchronize (e.g. new commits or new changes)

Either way, it changes the comment content.

Unlike dr.ci, these events are much less than `check_run` events, thus usually avg 1-2 writes per PR lifecycle.